File: backend/src/models/product.ts
import mongoose from 'mongoose'

export const ProductSchema = new mongoose.Schema({
    productId:{
        type: String, 
        require: true,
    },
    productName:{
        type: String, 
        require: true,
    },
    productDescription:{
        type: String,
        require: true,
    },
    productImage:{
        type: String,
        require: true
    },
    price:{
        type: Number,
        require: true
    },
    color:{
        type: String,
        require: true
    },
    size:{
        type: String,
        require: true
    },
    quantity:{
        type: Number,
        require: true
    },
    category:{
        type: String,
        require: true
    }
});

export default mongoose.model('Product', ProductSchema);

File: longdoo_frontend/lib/model/product.dart
class Product {
  final String id;
  final String productName;
  final double price;
  final double quantity;
  final String color;
  final String category;
  final String productDescription;
  final String productImage;

  Product.set({
    required this.id,
    required this.productName,
    required this.price,
    required this.quantity,
    required this.color,
    required this.category,
    required this.productDescription,
    required this.productImage,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product.set(
        id: json['_id'],
        productName: json['productName'],
        price: json['price'],
        quantity: json['quatity'],
        color: json['color'],
        category: json['category'],
        productDescription: json['productDescription'],
        productImage: json['productImage']);
  }

  Map<String, dynamic> toJson() {
    return {
      'productName': productName,
      'price': price,
      'quantity': quantity,
      'color': color,
      'category': category,
      'productDescription': productDescription,
      'productImage': productImage
    };
  }
}
